lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -25,3 +25,7 @@
  */
 add_action( 'switch_theme', 'wp_theme_clean_theme_json_cached_data' );
 add_action( 'start_previewing_theme', 'wp_theme_clean_theme_json_cached_data' );
+add_action( 'save_post_wp_global_styles', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'switch_theme', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'start_previewing_theme', 'gutenberg_clean_cached_stylesheet' );
+add_action( 'upgrader_process_complete', '_gutenberg_clean_cached_stylesheet_upon_upgrading_active_theme' );
```

Comment:
Some initial notes to invalidate the cache for settings & styles:

- Core data changes: when WordPress is updated?
- Block data changes & block registered:
  - Different blocks may be registered across requests or across different moments in time in the same request
  - What about 3rd parties filtering the block settings?
- Theme data changes: theme switch, theme preview, theme updated
- User data changes: wp_global_styles is saved, data is imported
- Listeners for existing filters change (theme.json data or block registration)
  - Plugins installed/uninstalled/updated.
  - Plugins updated: can modify the listeners
  - Drop-in files: how to detect this if we don't keep a list of listeners?